src/model-helpers/Document.js

```diff
@@ -0,0 +1,193 @@
+
+  let thing = buildThing(createThing({ name: docDesc.name }))
+    .addDate(RDF_PREDICATES.uploadDate, new Date())
+    .addStringNoLocale(RDF_PREDICATES.name, docDesc.name)
+    .addStringNoLocale(RDF_PREDICATES.identifier, docDesc.type)
+    .addDate(RDF_PREDICATES.endDate, docDesc.date)
+    .addStringNoLocale(RDF_PREDICATES.sha256, checksum)
+    .addStringNoLocale(RDF_PREDICATES.description, docDesc.description)
+    .addUrl(RDF_PREDICATES.url, documentUrl);
+
+  thing = addAdditionalInfo(thing, docDesc);
+  return thing.build();
+};
+
+export const parseDocument = (documentThing) => {
+  const uploadDate = getDate(documentThing, RDF_PREDICATES.uploadDate);
+  const name = getStringNoLocale(documentThing, RDF_PREDICATES.name);
+  const type = getStringNoLocale(documentThing, RDF_PREDICATES.identifier);
+  const endDate = getDate(documentThing, RDF_PREDICATES.endDate);
+  const checksum = getStringNoLocale(documentThing, RDF_PREDICATES.sha256);
+  const description = getStringNoLocale(documentThing, RDF_PREDICATES.description);
+  const fileUrl = getUrl(documentThing, RDF_PREDICATES.URL);
+  return { uploadDate, name, type, endDate, checksum, description, fileUrl };
+};
+
+const deleteRecursively = async (dataset, options) => {
+  const containedResourceUrls = getContainedResourceUrlAll(dataset);
+  const containedDatasets = await Promise.all(
+    containedResourceUrls.map(async (resourceUrl) => {
+      try {
+        return await getSolidDataset(resourceUrl, options);
+      } catch (e) {
+        // The Resource might not have been a SolidDataset;
+        // we can delete it directly:
+        await deleteFile(resourceUrl, options);
+        return null;
+      }
+    })
+  );
+  await Promise.all(
+    containedDatasets.map(async (containedDataset) => {
+      if (containedDataset === null) {
+        return;
+      }
+      await deleteRecursively(containedDataset, options);
+    })
+  );
+  await deleteSolidDataset(dataset, options);
+};
+
+export const saveDescription = async (docThing, dataset, session, docUrl) => {
+  const newDataset = setThing(dataset, docThing);
+  const name = getStringNoLocale(docThing, RDF_PREDICATES.name);
+  const doc = await saveSolidDatasetAt(`${docUrl}${name}.ttl`, newDataset, {
+    fetch: session.fetch
+  });
+  return doc;
+};
+
+export const createDocument = async (file, fileDescription, session, passUrl) => {
+  const { type, name } = fileDescription;
+  const docUrl = `${passUrl}Documents/${type}/${name}/`;
+  try {
+    await getSolidDataset(docUrl); // check to see if the file already exists
+  } catch {
+    const docThing = await docDescToThing(fileDescription, docUrl, file);
+    const doc = await saveDescription(docThing, createSolidDataset(), session, docUrl);
+    await saveFileInContainer(docUrl, file, {
+      fetch: session.fetch,
+      slugSuggestion: file.name
+    });
+    return parseDocument(getThingAll(doc)[0]);
+  }
+  throw Error('A file of this name already exists');
```

Comment:
I believe createDocument is the replacement for uploadDocument. If that's the case, think we should have the ability to update documents as well like what we had with updateDocument. There would be cases where people wish to store an updated file with the same name.

Reply:
added in `replaceDocument`